index/scorch/event.go

```diff
@@ -67,3 +67,10 @@ var EventKindMergeTaskIntroduction = EventKind(8)
 // EventKindPreMergeCheck is fired before the merge begins to check if
 // the caller should proceed with the merge.
 var EventKindPreMergeCheck = EventKind(9)
+
+// EventKindIndexStart is fired when Index() is invoked which
+// creates a new Document object from an interface using the index mapping.
+var EventKindIndexStart = EventKind(10)
+
+// EventKindIndex is fired when Index() completes.
```

Comment:
Then this should be called OnEnd, no?

Reply:
this is following the same naming convention as  `EventKindBatchIntroductionStart` and `EventKindBatchIntroduction` (for end) 
in cbft there is no handler for either `EventKindBatchIntroduction` or `EventKindIndex` so they are just dummy events added for API completeness

Reply:
actually never mind i have refactored the code now